### PR TITLE
feat: allow multiple authorized host options for SIWE

### PR DESCRIPTION
**Pull request by paolodamico**
SIWE apps may have multiple authorized domains. This allows handling the different options.

---

> [!NOTE]
> **Medium Risk**
> Touches SIWE host-authorization logic; incorrect matching could allow or block authentication for unintended domains. Scope is small but security-sensitive, so test coverage and edge cases (ports/schemes) matter.
> 
> **Overview**
> SIWE message parsing now accepts **multiple authorized URL options**: `SiweMessage::from_str_with_account` takes `authorized_urls: Vec<String>` and selects an `expected_authority` by matching the `querying_url` host/port against any provided authorized URL, rejecting if none match.
> 
> Tests are updated to pass URL lists and add coverage for matching from a mixed list of authorized domains, and `Cargo.lock` is refreshed (notably `rand` and `rustls-webpki` version bumps).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit d1aa14f1356607454d8932caf2efc250902671e0. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: bedrock/src/siwe/mod.rs [hunk 1]
```diff
@@ -391,27 +391,35 @@ impl SiweMessage {
...
-        }
+        let expected_authority: Authority =
+            {
+                let mut found = None;
+                for authorized_url in authorized_urls {
+                    let (expected_authority, _) = parse_authority(authorized_url)
+                        .map_err(|e| PrimitiveError::InvalidInput {
+                            attribute: "authorized_url".to_string(),
+                            error_message: e.to_string(),
+                        })?;
```
[the remark this answers is not in the document]
Comment (paolodamico) [reply bedrock/src/siwe/mod.rs:414]: ignoring malformed URLs could actually lead to harder to debug issues

**Review by Guardiola31337 — APPROVED** [review]
